RELEASE/scripts/autoscend/quests/level_9.ash

```diff
@@ -26,26 +26,45 @@ boolean LX_loggingHatchet()
 
 boolean L9_leafletQuest()
 {
-	if((my_level() < 9) || possessEquipment($item[Giant Pinky Ring]))
+	if(my_level() < 9)
 	{
 		return false;
 	}
-	if (isActuallyEd() || in_koe())
+	if(isActuallyEd() || in_koe())
 	{
 		return false;
 	}
-	if(auto_get_campground() contains $item[Frobozz Real-Estate Company Instant House (TM)])
+	
+	//check internal tracking on if we did leaflet or not. because mafia does not expose this tracking to us, yet.
+	//checking the resulting items is not reliable since they can be sold/lost/used/pulled from previous run/etc.
+	if(get_property("auto_L9LeafletQuestLastDone").to_int() == my_ascensions())
 	{
 		return false;
 	}
-	if(item_amount($item[Frobozz Real-Estate Company Instant House (TM)]) > 0)
+	
+	//get a [strange leaflet]
+	item leaflet = $item[strange leaflet];
+	if(closet_amount($item[strange leaflet]) > 0)
 	{
-		return false;
+		take_closet(1, $item[strange leaflet]);
+	}
+	if(available_amount($item[strange leaflet]) == 0)
+	{
+		council();
+		if(item_amount($item[strange leaflet]) == 0)
+		{
+			auto_log_debug("Tried to grab a [strange leaflet] from the council and it did not work... This needs fixing. skipping for now.");
+			return false;
+		}
 	}
+	
 	auto_log_info("Got a leaflet to do", "blue");
-	council();
 	cli_execute("leaflet");
-	use(1, $item[Frobozz Real-Estate Company Instant House (TM)]);
+	set_property("auto_L9LeafletQuestLastDone", my_ascensions());
+	if(auto_get_campground() contains $item[Newbiesport&trade; tent])
```

Comment:
I don't know if there are paths where you don't get the tent but this check should also use the house if the user has no dwelling. How you check that, I don't actually know (might need to ask on the mafia forum).

Reply:
As far as I know there is no way to just lookup what your dwelling is
I was going to refactor this bit further though, although I had not considered paths where you cannot get/use the tent though. So thanks for the reminder.

My plan was to (tomorrow) write a function `item myHouse()` that will go through all possible known dwelling types listed in the wiki and compare them all to get_campground() and return whichever one is installed in your camp.
If none of the known ones are found it returns $item[none]

then a second function auto_upgradeHouse()
to upgrade your housing which utilizes the prior function. 
replacing this line you quoted with `auto_upgradeHouse();`

Reply:
nevermind, there is apparently a function for this
https://wiki.kolmafia.us/index.php?title=Get_dwelling

Reply:
change made